onnxscript/evaluator.py

```diff
@@ -15,7 +18,62 @@
     InvalidGraph,
 )
 
-from onnxscript import irbuilder, tensor, utils
+from onnxscript import autocast, irbuilder, onnx_opset, tensor, utils, values
+
+
+class Evaluator(ABC):
+    """Base class for evaluation of ONNX ops.
+
+    The execution of onnxscript functions in eager-mode is dispatched to an Evaluator
+    instance (or, more precisely, to the eval method of the Evaluator instance).
+    The evaluator is expected to transform the input/output/attribute representation
+    supported by onnxscript to those expected by a particular backend.
+    """
+
+    def eval(self, schema, inputs, attributes):
+        closure = self.adapt_attributes(schema, attributes)
```

Comment:
I wonder if you should have a function adapt_outputs as well.

Reply:
done